src/sky.py

```diff
@@ -9,88 +9,64 @@
-        # number of seconds per in game minute (reference - stardew valley each
-        # minute is 0.7 seconds)
-        self.seconds_per_minute = 0.7
-        # gets the creation time in ticks
-        self.last_time = pygame.time.get_ticks()
-
-    def display(self, dt):
-        #   DAY / NIGHT CYCLE
-        current_time = pygame.time.get_ticks()
-        # if more than seconds_per_minute has passed, update clock
-        if (current_time - self.last_time > self.seconds_per_minute * 1000):
-            self.last_time = current_time
-            self.game_minute += 1
-
-            # minutes cycle every 60 in game minutes
-            if (self.game_minute > 59):
-                self.game_minute = 0
-                self.game_hour += 1
-            if (self.game_hour > 23):    # hours cycle every 24 in game hours
-                self.game_hour = 0
-
-            # Loop through each index of the current overlay color, then
-            # increment it by the above determined increments between noon and
-            # midnight
-            for index, value in enumerate(self.end_color):
-                if (self.game_hour >= 12):
-                    self.start_color[index] -= self.color_increment[index]
-                    if (self.start_color[index] < 0):
-                        self.start_color[index] = 0
-                else:
-                    self.start_color[index] += self.color_increment[index]
-                    if (self.start_color[index] > 255):
-                        self.start_color[index] = 255
-
-        self.full_surf.fill(self.start_color)
-        self.display_surface.blit(
-            self.full_surf, (0, 0), special_flags=pygame.BLEND_RGBA_MULT)
-
-    def set_time(self, hours, minutes):
-        self.game_hour = hours
-        self.game_minute = minutes
-        num_of_minutes = hours * 60 + minutes
+        self.game_time = game_time
 
-        # if number of minutes is less than noon, add increments to color
-        # if number of minutes is greater than noon, subtract increments to
         # color
-        if (num_of_minutes < 720):
-            new_color = [self.end_color[0], self.end_color[1],
-                         self.end_color[2]]         # darkest color is midnight
-            for index, value in enumerate(self.end_color):
-                new_color[index] += self.color_increment[index] * \
-                    num_of_minutes
+        self.colors = {
+            '6' : (160, 187, 255),
+            '12' : (255, 255, 255),
+            '18' : (255, 240, 234),
+            '20' : (255, 219, 203),
+            '24' : (38, 101, 189)
+        }
+        self.color = self.get_color()
+
+    def get_color(self):
+
+        # get time
+        hour, minute = self.game_time.get_time()
+        precise_hour = hour + minute / 60
+
+        # find nearest hours in self.colors
+        color_index = 0
+        for index in range(len(self.colors)):
+            color_hour = int(list(self.colors.keys())[index])
```

Comment:
`list(self.color.keys())` is redundant. `list(self.colors)` will achieve the same effect. Also, you might want to use an equality comparison instead of using indexes here.

Reply:
Solved